torch/csrc/jit/runtime/register_distributed_ops.cpp

```diff
@@ -63,10 +68,13 @@ RegisterOperators reg_rpc_ops(
          },
          aliasAnalysisFromSchema()),
      Operator(
-         "aten::owner(RRef(t) self) -> int",
+         "aten::owner(RRef(t) self) -> __torch__.torch.classes.dist_rpc.WorkerInfo",
          [](Stack& stack) {
            auto rref = pop(stack).toRRef();
-           push(stack, rref->owner());
+           push(
+               stack,
+               torch::make_custom_class<distributed::rpc::WorkerInfo>(
```

Comment:
why we need to make a new instance here? can't we do `RRefContext::getInstance().agent()->getWorkerInfo(rref_->owner());`?

Reply:
IValue constructor can take `intrusive_ptr<WorkerInfo>`, but `RRefContext::getInstance().agent()->getWorkerInfo(..)` returns a const ref, `const WorkerInfo&`. 

We don't want it to return a `intrusive_ptr<WorkerInfo>`, because the `allWorkerInfo_` vector is a static setup on constructing RpcAgent, we must let RpcAgent decide it's lifetime.

Similarly, Python side API, `rref.owner()`, asks for this data as well, so we make a copy of this const WorkerInfo, and let Python holds the shared_ptr to this WorkerInfo copy, and Python and decide the lifetime of the copy arbitrarily, without being affected by destruction of RpcAgent and it's contained `allWorkerInfo_` vector.

Reply:
I see, that make sense, but when the rpcAgent shutdowns, won't the workerInfos not valid anymore, in this case user is holding a non-valid workerInfo, and user might later pass it with a newly created rpcAgent that holds a different worker list, will that cause problem?

Reply:
I only image application code holds it for printing debugging related info. e.g. we have rref leak detection code, which could make use of this information.

Your concern is possibly true, since WorkerInfo are just id + name. In new session, the id->name mapping could be different. This will take some effort to fix. 

We didn't prioritize because it's never a real use case. If we will have a new distributed training session, workers are usually new sub-process. Highly impossible to be shared about WorkerInfo copy.